tests/optimize/test_backtesting_adjust_position.py

```diff
@@ -197,3 +197,130 @@ def test_backtest_position_adjustment_detailed(default_conf, fee, mocker, levera
+    pair = 'XRP/USDT:USDT'
+    row = [
+        pd.Timestamp(year=2023, month=1, day=1, hour=5, minute=0),
+        2.1,  # Open
+        2.2,  # High
+        1.9,  # Low
+        2.1,  # Close
+        1,  # enter_long
+        0,  # exit_long
+        0,  # enter_short
+        0,  # exit_short
+        '',  # enter_tag
+        '',  # exit_tag
+        ]
+
+    backtesting.strategy.leverage = MagicMock(return_value=10.0)
+    trade = backtesting._enter_trade(pair, row=row, direction='long')
+    assert trade
+
+    print(f'trade.liquidation_price: {trade.liquidation_price}')
+
+    # position adjust creating exit order
+    def custom_entry_price(proposed_rate, **kwargs):
+        return proposed_rate * 0.95
+    
+    backtesting.strategy.custom_entry_price = custom_entry_price
+    backtesting.strategy.adjust_trade_position = MagicMock(return_value=5)
+
+    # wait for stoploss trigger event
+    row_sl_tgrd = [
+        pd.Timestamp(year=2023, month=1, day=1, hour=5, minute=30),
+        1.1,  # Open
+        1.2,  # High
+        0.9,  # Low
+        1.11,  # Close
+        0,  # enter_long
+        0,  # exit_long
+        0,  # enter_short
+        0,  # exit_short
+        '',  # enter_tag
+        '',  # exit_tag
+        ]
+    res = backtesting._check_trade_exit(trade, row_sl_tgrd, row_sl_tgrd[0].to_pydatetime())
+
+    # TODO count and validate that there is a open entry order
+    print(res)
+    print(f'exit reason : {res.exit_reason}')
+
+    # TODO This should FAIL curently as exits is not triggered if trade.has_open_orders
+    assert res
+    assert res.exit_reason == ExitType.STOP_LOSS.value
+
+def test_backtest_liquidation_with_open_exit_order(default_conf, fee, mocker,) -> None:
+    default_conf['use_exit_signal'] = False
+    default_conf['max_open_trades'] = 10
+    default_conf['stake_amount'] = 10
+    mocker.patch(f'{EXMS}.get_fee', fee)
+
+    backtesting = Backtesting(default_conf)
+    backtesting._set_strategy(backtesting.strategylist[0])
+    pair = 'XRP/USDT:USDT'
+    row = [
+            pd.Timestamp(year=2023, month=1, day=1, hour=5, minute=0),
+            2.1,  # Open
+            2.2,  # High
+            1.9,  # Low
+            2.1,  # Close
+            1,  # enter_long
+            0,  # exit_long
+            0,  # enter_short
+            0,  # exit_short
+            '',  # enter_tag
+            '',  # exit_tag
+            ]
+
+    # perform entry with high leverage
+    backtesting.strategy.leverage = MagicMock(return_value=40.0)
+
+    # TODO how can I edit the stoploss to allow liquidation to occur?
+    backtesting.strategy.stop_loss_abs = MagicMock(return_value=0.1)
+    backtesting.strategy.stop_loss_ratio = MagicMock(return_value=-0.9)
+    trade = backtesting._enter_trade(pair, row=row, direction='long')
+    assert trade
+
+    print(f'trade.liquidation_price: {trade.liquidation_price}')
+    # TODO QUESTION Why leverage stay at 1.0 ?
```

Comment:
@xmatthias could you provide me the proper way to edit leverage and stoploss in order to create the test scenario properly?

Reply:
your test runs on spot markets (the default for `default_conf`).
Futures markets are "opt in" ... so you need to re-set trading_mode to futures for leverage to not be ignored.

I'd also use `default_conf_usdt` for new tests ... doing tests with BTC  feels a bit awkward if you consider it's volatility (most older tests assume a price of 10-15k - which nowadays seems odd) - i always found that to help with trying to do the calculations manually to ensure the test actually makes sense.